storm-client/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -248,11 +252,8 @@ public static void submitTopologyAs(String name, Map<String, Object> topoConf, S
         try {
             String serConf = JSONValue.toJSONString(topoConf);
             try (NimbusClient client = NimbusClient.getConfiguredClientAs(conf, asUser)) {
-                if (topologyNameExists(name, client)) {
-                    throw new RuntimeException("Topology with name `" + name + "` already exists on cluster");
-                }
-                if (!Utils.isValidKey(name)) {
```

Comment:
why was this removed?

Reply:
There are two reasons:
1. Moved the name check to the very beginning of this submit function. 
2. Replace it with `Utils.validateTopologyName(name);` so we have the same check on client and server side.